src/switch_core_session.c

```diff
@@ -1459,6 +1459,83 @@ SWITCH_DECLARE(uint32_t) switch_core_session_flush_and_publish_private_events(sw
 	return x;
 }
 
+SWITCH_DECLARE(uint32_t) switch_core_session_flush_and_publish_private_media_events(switch_core_session_t *session)
+{
+	switch_status_t status = SWITCH_STATUS_FALSE;
+	switch_queue_t *queue_pri, *queue;
+	int x = 0;
+	void *pop;
+	
+	if (session->private_event_queue) {
+		switch_memory_pool_t *pool = switch_core_session_get_pool(session);
+
+		if (switch_queue_size(session->private_event_queue_pri)) {
+			switch_queue_create(&queue_pri, SWITCH_EVENT_QUEUE_LEN, pool);
+			while ((status = (switch_status_t) switch_queue_trypop(session->private_event_queue_pri, &pop)) == SWITCH_STATUS_SUCCESS) {
+				if (pop) {
+					switch_event_t *interrupt_event = NULL;
+					switch_event_t *command_event = (switch_event_t *) pop;
+					const char *app_name = switch_event_get_header(command_event, "execute-app-name");
+
+					// filter only playback and TTS cmds
+					if (!strcasecmp(app_name, playback_cmd) || !strcasecmp(app_name, speak_cmd) || !strcasecmp(app_name, say_cmd)) {
```

Comment:
This is still not flexible, we might need to update this in the futre when they decide to stop a media commands such as `telnyx_play_and_get_digits` or `play_and_get_digits` or any similar functions. See my comment in the header file